### PR TITLE
Sensor Refactor (messed up prev branch 'sensor_refactor_python_version_3')

### PR DESCRIPTION
**Pull request by jrreyna**
## Motivation and Context

Refactoring sensor code in python to mirror what the C++ code does. Continuing work from pull requests #1118 and #1151 (and this improves upon the work in the branch correlating to PRs #1921 and #1849).

#1118 introduced SensorSuites and SubtreeSensorSuites in scene nodes.

#1151 attempted to deprecate simulator.Sensor and SensorSuites.

I think in future, sensors should have references to their agent (if not None) and their scene node. Agents should have references to their node and sensor suite, and scene nodes should have references to their agents and maybe their own adapted sensor suite. Those are just simple changes we can make without rearranging the entire sensor architecture

## How Has This Been Tested

Locally and CI tests. Works on my machine and passes all tests, but CI tests upon pushing the code fail because there are some bugs in it.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jrreyna** [comment]
Code passes CI tests on my machine

File: src/esp/bindings/SensorBindings.cpp [hunk 1]
```diff
@@ -248,7 +250,102 @@ void initSensorBindings(py::module& m) {
...
+                           esp::sensor::SensorType::Depth) {
+                  py::setattr(
+                      handle, "__obs_buffer",
+                      torch.attr("empty")(
+                          (py::int_(self.specification()->resolution[0]),
+                           py::int_(self.specification()->resolution[1])),
+                          "dtype"_a = torch.attr("float32"),
+                          "device"_a = torch.attr("device")(
+                              py::str("cuda"), py::int_(gpuDevice))));
+                } else {
```
Comment (Skylion007) [inline src/esp/bindings/SensorBindings.cpp:281]: This isn't future proof if we introduce a new SensorType, we should raise an error potentially if it's not an expected SensorType.

Comment (jrreyna) [reply src/esp/bindings/SensorBindings.cpp:281]: I might actually just remove this for now. This was a carry over from the PR by that other former employee that I continued. I don't believe I am even using these bindings right now

Comment (jrreyna) [reply src/esp/bindings/SensorBindings.cpp]: @Skylion007, like I said, I may just remove this section, but if I don't, would this be sufficient error catching?

**Pull request closed without merging**